### PR TITLE
Move log library from logrus to src-d/go-log

### LINKED ISSUE #262: Standardise logging library in borges (gopkg.in/src-d/go-logger.v1)
if something is missed try to enhance gopkg.in/src-d/go-logger.v1 library instead of replace it by any thirdparty (log15, logrus, ...)

### PR DESCRIPTION
**Pull request by mcarmonaa**
Replaced `logrus` by [src-d/go-log](https://github.com/src-d/go-log)

Closes #262 

Signed-off-by: Manuel Carmona <manu.carmona90@gmail.com>

### CODE REVIEW

**Comment by kuba--** [comment]
Wow! this almost looks like a big, manual revert to what I did y'day ;)
The only question I have (after we switch form logrus to go-log) do we still have file and line numbers?
Because afaik go-log is based on top of logrus and logrus natively does not support `file:line` in logs. It's only available through _hooks_

**Comment by mcarmonaa** [comment]
@kuba-- good point, I didn't put attention on that. Indeed, we don't have `file: line` now on the logs :disappointed: , should I open an issue to keep it in mind so we can add that functionality to [src-d/go-log](https://github.com/src-d/go-log) later on?

**Comment by kuba--** [comment]
Regarding `filename:line` in `go-log`, I've sent a PR: https://github.com/src-d/go-log/pull/2

File: archiver.go [hunk 1]
```diff
@@ -128,34 +129,33 @@ func (a *Archiver) do(log *logrus.Entry, j *Job) (err error) {
...
+				}).Error(err, "error setting repo as failed")
 			}
 
 			err = ErrFatal.New(rcv, debug.Stack())
 		}
 	}()
 
 	if err := a.canProcessRepository(r, &now); err != nil {
-		log.WithFields(logrus.Fields{
+		l.New(log.Fields{
```
Comment (mcuadros) [inline archiver.go]: This is not the intended use for `New`. 
New is expected to be used for pass a `logger` to the context, and not to directly inline.

File: archiver.go [hunk 2]
```diff
@@ -128,34 +129,33 @@ func (a *Archiver) do(log *logrus.Entry, j *Job) (err error) {
...
 		if rcv := recover(); rcv != nil {
-			log.WithField("error", rcv).Error("panic while processing repository")
-
+			l.Error(fmt.Errorf("%v", rcv), "panic while processing repository")
 			r.FetchErrorAt = &now
-			if sErr := a.Store.UpdateFailed(r, model.Pending); sErr != nil {
-				log.WithFields(logrus.Fields{
+			if err := a.Store.UpdateFailed(r, model.Pending); err != nil {
+				l.New(log.Fields{
 					"err": err,
```
Comment (mcuadros) [inline archiver.go]: The err is repeated.

**Comment by mcuadros** [comment]
As I wrote https://github.com/src-d/borges/pull/261#issuecomment-383557288, the logging should be blocked, until a guide about logging and go-log is released. 

Most likely border logging will be implemented by @smola or me, to learn about use cases, and write the guide.

Sorry :/

**Pull request closed without merging**